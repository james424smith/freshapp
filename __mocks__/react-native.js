import * as ReactNative from "react-native";

export * from "react-native";
export const Vibration = { vibrate: jest.fn() };
export const Adnimated = { spring: jest.fn() };

export default Object.setPrototypeOf(
  {
    Vibration,
    Adnimated,
  },
  ReactNative
);
