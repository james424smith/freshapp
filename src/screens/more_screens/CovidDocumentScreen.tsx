import React, { useEffect } from "react";
import { ScrollView, View, StyleSheet } from "react-native";
import PdfRender from "../../components/PdfRender";
import { useSelector, useDispatch } from "react-redux";
import { getCovidDocument } from "../../redux/actions";
import { IRootReducerType } from "../../redux/reducers";
import LoadingScreen from "../../components/LoadingScreen";

const logoSize = 100;

const styles = StyleSheet.create({
  root: { flex: 1 },
  scrollViewContainer: {
    flexGrow: 1,
  },
});

const CovidDocumentScreen = () => {
  const covidDocumentFile = useSelector<IRootReducerType, string>(
    ({ covidDocument }) => covidDocument.covidDocument
  );

  const isLoading = useSelector<IRootReducerType, boolean>(
    ({ covidDocument }) => covidDocument.loader
  );

  const dispatch = useDispatch();

  useEffect(() => {
    dispatch(getCovidDocument());
  }, [dispatch]);

  const pdf = covidDocumentFile;

  if (isLoading) {
    return (
      <View style={styles.root}>
        <LoadingScreen logoWidthSize={logoSize} logoHeightSize={logoSize} />
      </View>
    );
  }

  return (
    <ScrollView contentContainerStyle={styles.scrollViewContainer}>
      {pdf ? <PdfRender document={pdf} resourceType={"base64"} /> : <View />}
    </ScrollView>
  );
};

export default CovidDocumentScreen;
