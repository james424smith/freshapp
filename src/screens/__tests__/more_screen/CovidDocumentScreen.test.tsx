import React from "react";
import { render, cleanup } from "@testing-library/react-native";
import CovidDocumentScreen from "../../more_screens/CovidDocumentScreen";
import { Provider } from "react-redux";
import configureMockStore from "redux-mock-store";

describe("CovidDocumentScreen snapshot test", () => {
  afterEach(() => {
    cleanup();
    jest.clearAllMocks();
    jest.restoreAllMocks();
  });

  it("should render CovidDocumentScreen when loader is false ", () => {
    const mockedState = {
      covidDocument: {
        covidDocument: "1234",
        loader: false,
      },
      network: { isConnected: true },
    };
    const mockedStore = configureMockStore();

    const { toJSON } = render(
      <Provider store={mockedStore(mockedState)}>
        <CovidDocumentScreen />
      </Provider>
    );

    expect(toJSON()).toMatchSnapshot();
  });
  it("should render CovidDocumentScreen when loader is false  and covidDocument is undefined", () => {
    const mockedState = {
      covidDocument: {
        covidDocument: undefined,
        loader: false,
      },
      network: { isConnected: true },
    };
    const mockedStore = configureMockStore();

    const { toJSON } = render(
      <Provider store={mockedStore(mockedState)}>
        <CovidDocumentScreen />
      </Provider>
    );

    expect(toJSON()).toMatchSnapshot();
  });
  it("should render CovidDocumentScreen when loader is true ", () => {
    const mockedState = {
      covidDocument: {
        covidDocument: "1234",
        loader: true,
      },
      network: { isConnected: true },
    };
    const mockedStore = configureMockStore();

    const { toJSON } = render(
      <Provider store={mockedStore(mockedState)}>
        <CovidDocumentScreen />
      </Provider>
    );

    expect(toJSON()).toMatchSnapshot();
  });
});
