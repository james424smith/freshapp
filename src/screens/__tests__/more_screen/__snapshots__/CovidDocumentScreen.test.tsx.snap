// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CovidDocumentScreen snapshot test should render CovidDocumentScreen when loader is false  1`] = `
<RCTScrollView
  contentContainerStyle={
    Object {
      "flexGrow": 1,
    }
  }
>
  <View>
    <View
      style={
        Object {
          "flex": 1,
        }
      }
    >
      <PDFView
        fadeInDuration={10}
        onError={[Function]}
        onLoad={[Function]}
        onPageChanged={[Function]}
        onScrolled={[Function]}
        resource="1234"
        resourceType="base64"
        style={
          Object {
            "flex": 1,
            "width": 750,
          }
        }
        textEncoding="utf-8"
        urlProps={Object {}}
      />
    </View>
  </View>
</RCTScrollView>
`;

exports[`CovidDocumentScreen snapshot test should render CovidDocumentScreen when loader is false  and covidDocument is undefined 1`] = `
<RCTScrollView
  contentContainerStyle={
    Object {
      "flexGrow": 1,
    }
  }
>
  <View>
    <View />
  </View>
</RCTScrollView>
`;

exports[`CovidDocumentScreen snapshot test should render CovidDocumentScreen when loader is true  1`] = `
<View
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "flex": 1,
        "justifyContent": "center",
        "marginBottom": 112.5,
      }
    }
  >
    <Image
      source={Object {}}
      style={
        Object {
          "height": 100,
          "width": 100,
        }
      }
    />
  </View>
</View>
`;
